homeassistant/components/netatmo/climate.py

```diff
@@ -441,12 +459,31 @@ async def _async_service_set_preset_mode_with_end_datetime(
             mode=PRESET_MAP_NETATMO[preset_mode], end_time=end_timestamp
         )
         _LOGGER.debug(
-            "Setting %s preset to %s with optional end datetime to %s",
+            "Setting %s preset to %s with optional end datetime %s",
             self._room.home.entity_id,
             preset_mode,
             end_timestamp,
         )
 
+    async def _async_service_set_temperature(self, **kwargs: Any) -> None:
```

Comment:
There is already a service for setting temperature so this only makes sense if end_datetime is not optional.

Reply:
See `_async_service_set_preset_mode_with_end_datetime()` which had the same comments in a previous PR

Reply:
That had confused me a little as to why the end date time was required when it is not for the API. 

I understand wanting to differentiate between `climate.set_temperature` and `netatmo.set_temperature`. It made more sense to me to have it as `set_temperature` as that is what it is primarily doing, just with some additional optional functionality which the netatmo ecosystem provides. This is currently only a single additional argument but that might not always be the case and there may be many combinations of available parameters in future. Having it optional also provides the user with the guidance that it will use the netatmo default if not provided. The set temperature service currently does this silently (3 hours).

I can change it if you still prefer. I certainly don't like it being inconsistent with the set preset service. It just felt like the more sensible name to me for future maintenance and possible expansion.

I've looked through the other PR. As far as I can tell the naming is the only issue so far. I'm not clear on the process so I've marked it as ready for review again for this comment to be considered.